diff --diff a/lib/curl_threads.h b/lib/curl_threads.h
--- a/lib/curl_threads.h
+++ b/lib/curl_threads.h
@@ -26,7 +26,7 @@
 #include "curl_setup.h"
 
 #ifdef USE_MUTEX
-#ifdef _WIN32
+#if defined(_WIN32) && (_WIN32_WINNT >= _WIN32_WINNT_VISTA)   /* [nscurl] original implementation using native CONDITION_VARIABLE (targeting vista)*/
 #  define CURL_THREAD_RETURN_T   DWORD
 #  define CURL_STDCALL           WINAPI
 #  define curl_mutex_t           CRITICAL_SECTION
@@ -40,6 +40,25 @@
 #  define curl_cond_t            CONDITION_VARIABLE
 #  define Curl_cond_init(c)      InitializeConditionVariable(c)
 #  define Curl_cond_destroy(c)   (void)(c)
+#elif defined(_WIN32) && (_WIN32_WINNT < _WIN32_WINNT_VISTA)   /* [nscurl] custom implementation of CONDITION_VARIABLE (targeting nt4..xp) */
+#  define CURL_THREAD_RETURN_T   DWORD
+#  define CURL_STDCALL           WINAPI
+#  define curl_mutex_t           CRITICAL_SECTION
+#  define curl_thread_t          HANDLE
+#  define curl_thread_id_t       DWORD
+#  define curl_thread_t_null     (HANDLE)0
+#  define Curl_mutex_init(m)     InitializeCriticalSectionEx(m, 0, 1)
+#  define Curl_mutex_acquire(m)  EnterCriticalSection(m)
+#  define Curl_mutex_release(m)  LeaveCriticalSection(m)
+#  define Curl_mutex_destroy(m)  DeleteCriticalSection(m)
+struct curl_cond_struct {
+    HANDLE event;
+    int pending_signals;
+    CRITICAL_SECTION signals_mutex;
+};
+typedef struct curl_cond_struct curl_cond_t;    /* nscurl: CONDITION_VARIABLE replacement */
+void Curl_cond_init(curl_cond_t *c);            /* nscurl: custom implementation */
+void Curl_cond_destroy(curl_cond_t *c);         /* nscurl: custom implementation */
 #elif defined(HAVE_THREADS_POSIX)
 #  define CURL_THREAD_RETURN_T   unsigned int
 #  define CURL_STDCALL
