diff --git a/lib/easy_lock.h b/lib/easy_lock.h
--- a/lib/easy_lock.h
+++ b/lib/easy_lock.h
@@ -27,7 +27,8 @@
 
 #define GLOBAL_INIT_IS_THREADSAFE
 
-#ifdef _WIN32
+/* [nscurl] original implementation using SRWLOCK (targeting vista) */
+#if defined(_WIN32_WINNT) && (_WIN32_WINNT >= _WIN32_WINNT_VISTA)
 
 #define curl_simple_lock      SRWLOCK
 #define CURL_SIMPLE_LOCK_INIT SRWLOCK_INIT
@@ -73,6 +74,8 @@
 #elif defined(__aarch64__)
       /* NOLINTNEXTLINE(portability-no-assembler) */
       __asm__ volatile("yield" ::: "memory");
+#elif defined(_WIN32)
+      Sleep(1);
 #elif defined(HAVE_SCHED_YIELD)
       sched_yield();
 #endif
