diff --git a/lib/curlx/winapi.c b/lib/curlx/winapi.c
--- a/lib/curlx/winapi.c
+++ b/lib/curlx/winapi.c
@@ -106,6 +106,8 @@
 
 #ifndef WITHOUT_LIBCURL
 
+#if (_WIN32_WINNT >= _WIN32_WINNT_VISTA)    /* [nscurl] original implementation using BCrypt calls */
+
 #include <bcrypt.h>
 #ifndef STATUS_SUCCESS
 #define STATUS_SUCCESS ((NTSTATUS)0x00000000L)
@@ -121,6 +123,30 @@
 
   return CURLE_OK;
 }
+
+#else   /* [nscurl] legacy implementation using CryptoAPI calls (targets nt4..xp) */
+
+#include <wincrypt.h>
+
+CURLcode curlx_win32_random(unsigned char *entropy, size_t length)
+{
+  CURLcode ret = CURLE_FAILED_INIT;
+  HCRYPTPROV provider = 0;
+
+  memset(entropy, 0, length);
+
+  if(CryptAcquireContext(&provider, NULL, NULL, PROV_RSA_FULL, CRYPT_VERIFYCONTEXT)) {
+    if(CryptGenRandom(provider, (DWORD)length, entropy)) {
+      ret = CURLE_OK;
+    }
+    CryptReleaseContext(provider, 0);
+  }
+
+  return ret;
+}
+
+#endif /* [nscurl] */
+
 #endif /* WITHOUT_LIBCURL */
 
 #endif /* _WIN32 */
